src/porepy/compositional/compositional_mixins.py

```diff
@@ -61,9 +66,88 @@ def _get_surrogate_factory_as_property(
     )
 
 
+# NOTE The following methods are outsourced for compatibility reasons.
```

Comment:
I am open to put this into the base `SolutionStrategy` like the methods Eirik mentioned.
Originally I was hesitant to do that because two of them are only relevant for models with flash calculations.

Reply:
Let's move them into the solution strategy, then.

Reply:
I moved the inquires about eliminated reference phase and component to the base `SolutionStrategy`, because it has a general fluid now and it is therefore compatible with all models.

I moved the fractional flow inquire to `compositional_flow`, because it is only used there, and I removed imports in packages on any level since not required.

The inquires about the equilibrium type I will leave for now because they are required to determine the DOFs for CF.
Added a note to revisit their placing once the flash and CFLE are ready to be merged.

If this is fine with you, please resolve.